### ISSUE TITLE
$model->hasManyRelation()->saveMany($models) fails, tries to defer binding always.

### ISSUE THREAD

**Issue by pfrend** [octobercms/october#5168]
The saveMany() method in Illuminate\Database\Eloquent\Relations\HasOneOrMany does not function correctly.

The problem is it calls array_walk with the current class instance and the save method as array params:
array_walk($models, array($this, 'save'));

This means the save method is called with the index of the current array_walk iteration as the second param, which then gets passed as the $sessionKey in the October specific implementation of $model->save().

Will submit a PR. 

Basically I believe we just need to override saveMany() as has been done with ->save(), and use a simple loop instead of array_walk, with the optional $sessionKey passed to save().
